lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,344 @@
+   * seemed like the less bad option.
+   */
+  elFocusedOnMostRecentRender.current = typeof document === 'undefined' ? null : document.activeElement;
+
+  const checkForRemovedFocusedEl = useCallback(function checkForRemovedFocusedEl(mutations: MutationRecord[]) {
+    const nodeContainedFocus = (el: Node) => el.contains(elFocusedOnMostRecentRender.current),
+        nodeListContainedFocus =
+            pipe<[NodeList], Node[], boolean>(Array.from, any(nodeContainedFocus)),
+        focusedChildWasRemoved =
+            any(pipe(prop('removedNodes'), nodeListContainedFocus), mutations);
+
+    if (focusedChildWasRemoved) {
+      focusTextInput();
+    }
+  }, []);
+
+  useMutationObserver(dropdownRef, checkForRemovedFocusedEl, { childList: true });
+
+  function handleOnClick(match: DataItem<T>) {
+    onSelect(match);
+    focusTextInput();
+    setElToFocusId('');
+    setFocusableBtnIndex(null);
+    setShowDropdown(false);
+  }
+
+  function handleElInView(el: HTMLElement) {
+    const bounding = el.getBoundingClientRect();
+    const dropdownMenu = dropdownRef.current;
+
+    //this check if the dropdown menu is not in the document view
+    const isInView = bounding.top >= 0 && bounding.left >= 0 &&
+      bounding.bottom <= (document.documentElement.clientHeight)
+      && bounding.right <= (document.documentElement.clientWidth);
+
+    if (!isInView) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+
+    //this check if the option is not visible inside the dropdown menu
+    if (dropdownMenu &&
+      dropdownMenu.scrollTop + dropdownMenu.offsetHeight <
+      el.offsetTop + el.offsetHeight
+    ) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+    else if (dropdownMenu && dropdownMenu.scrollTop > el.offsetTop + 4) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+  }
+
+  return (
+    /*eslint-disable-next-line jsx-a11y/no-static-element-interactions*/
+    <div ref={ref}
+         className={className}
+         onFocus={handleComponentFocus}
+         onBlur={handleComponentBlur}
+         onMouseDown={handleMouseDown}
+         { ...attrs }>
+      <NxTextInput role="combobox"
+                   ref={inputRef}
+                   id={inputId}
+                   isPristine
+                   {...inputProps}
+                   className={inputClassName}
+                   value={searchText}
+                   onChange={handleFilterChange}
+                   disabled={disabled || undefined}
+                   onKeyDown={handleKeyDown}
+                   aria-autocomplete={autoComplete ? 'both' : 'list'}
+                   aria-expanded={showDropdown}
+                   aria-controls={dropdownId}
+                   aria-activedescendant={elToFocusId}
+                   aria-required={ariaRequired ?? undefined}
+                   aria-describedby={ariaDescribedBy ?? undefined}/>
```

Comment:
Per the [spec](https://www.w3.org/TR/wai-aria-1.2/#aria-autocomplete) for `aria-autocomplete`, the input should have `aria-haspopup="listbox"`, when the dropdown is in fact a `listbox`.

It's worth mentioning however that the spec seems to somewhat contradict itself, as the part about `aria-haspopup` itself recommends only using the attribute when there is a _visible_ indicator that a popup is available (e.g. a down arrow in the combobox) which we don't have

Reply:
> Per the [spec](https://www.w3.org/TR/wai-aria-1.2/#aria-autocomplete) for aria-autocomplete, the input should have aria-haspopup="listbox", when the dropdown is in fact a listbox.

Oh wait, the part of the spec about the `combobox` role says that `aria-haspopup="listbox" is set by default for that role